### DISCUSSION TITLE
Is it possible to rearrange the Field order in Alerts?

### DISCUSSION THREAD

**Discussion by TFWol** [Security-Onion-Solutions/securityonion#7900] [Unsupported Versions] [answered]
Is there any way to change the default order **Fields** are displayed in the **Alerts** view?
I'm trying to prevent alphabetical sorting.

Screenshot from the SecoOnion docs website and what I'm trying to do (with any fields though):
![image](https://user-images.githubusercontent.com/9045213/166829477-83cffc96-89bd-408a-b320-b5b00074b071.png)

I was able to change the columns based on the [documentation ](https://docs.securityonion.net/en/2.3/soc-customization.html#alerts), but I wasn't able to find a reference to the Fields in the detailed view.
I attempted to search through Kibana's docs… but I lost hope from the complexity.
There were instructions on sorting values, creating calculated columns, mapping tags, and so forth. 
In the end I still couldn't even tell if it was possible.
	
Some of the posts I tried gleaning information from weren't all that helpful (even more confusing or not answered):
 - https://discuss.elastic.co/t/how-to-re-arrange-or-sort-fields-of-source-in-kibana-by-default-kibana-is-sorting-alphabetically/212780/2
 - https://discuss.elastic.co/t/how-to-organize-scripted-fields-in-order/211216
 - https://stackoverflow.com/questions/33382734/fields-not-getting-sorted-in-alphabetical-order-in-elasticsearch
 - https://stackoverflow.com/questions/31597847/how-to-return-fields-in-correct-order-for-an-elasticsearch-query

It feels like I have to learn to build a car before I can adjust the AC.

**Comment by dougburks** [comment, marked as answer]
I don't believe there is an easy way to rearrange the fields like that. What is the end goal that you're trying to achieve by rearranging fields? Are you just trying to get more visibility into certain fields like `destination.geo.country_name`? Have you tried the `Group By` option on the context menu as shown in your screenshot?

**Reply by TFWol** [reply]
Drat, too bad Kibana tweaks can't be easy.
My main goal was to put `network.data.decoded` up top so I could just glance through quickly instead of scrolling down every time.
Setting it as a column gets too unwieldy.

**Reply by dougburks** [reply]
> too bad Kibana tweaks can't be easy.

Just to make sure we're on the same page, your title and screenshot are referring to our SOC Alerts interface (not Kibana).

In SOC Alerts, you can groupby `network.data.decoded` like this:
![image](https://user-images.githubusercontent.com/1659467/167127758-3f56da37-70ef-402f-b19f-fcf8cbb3ad35.png)

You can also search for NIDS alerts in our Hunt interface and then groupby `network.data.decoded` like this:
![image](https://user-images.githubusercontent.com/1659467/167128090-4f6487a6-e095-47d6-971c-d422e5d47561.png)

**Reply by TFWol** [reply]
> Just to make sure we're on the same page, your title and screenshot are referring to our SOC Alerts interface (not Kibana)

Yes, sorry I'm referring to the Alerts screen. I had assumed the change would have something to do with the Kibana back-end since a lot of things seemed tied to it.

> You can also search for NIDS alerts in our Hunt interface and then groupby network.data.decoded

Yeah, that's the unwieldy look I was referring to. I was shooting for a dense view, like the original screenshot, but with the fields ordered a particular way with `network.data.decoded` word-wrapped like it normally would be.

Pretty much like this, but whatever fields in whatever order:
![image](https://user-images.githubusercontent.com/9045213/167136855-7e326555-3537-4ec9-8f83-32ed5ad91f0c.png)

Although based on what you're saying, I'd probably just have to get used to a different kind of view.

**Reply by dougburks** [reply]
> Yes, sorry I'm referring to the Alerts screen. I had assumed the change would have something to do with the Kibana back-end since a lot of things seemed tied to it.

To be clear, Kibana is not a back-end. It is a front-end web interface that connects to Elasticsearch on the back-end (similar to how our SOC interface is a front-end that connects to Elasticsearch on the back-end).

> Although based on what you're saying, I'd probably just have to get used to a different kind of view.

Depending on your exact use case, you might also consider pivoting to PCAP. This would show you the entire stream and not just the small snippet that is shown in `network.data.decoded`.

For more information, please see:
https://docs.securityonion.net/en/2.3/pcap.html#pcap

**Reply by TFWol** [reply]
> To be clear, Kibana is not a back-end. It is a front-end web interface that connects to Elasticsearch on the back-end (similar to how our SOC interface is a front-end that connects to Elasticsearch on the back-end).

Got it. Looks like I was using the terms interchangeably.

> Depending on your exact use case, you might also consider pivoting to PCAP. This would show you the entire stream and not just the small snippet

I've used that view for traffic that looked concerning to me, although I like to have things easy to eyeball since I pretty much use SecOnion to keep tabs on what applications may be sending out.
